File: Sorting Algorithms/Bubble Sort/Diff Approaches/Python/bubble_sort_alt1.py
arr = [1,5,2,7,5,8,2,5,8]

print("Array before getting sorted: ")
print(arr)




def bubble_sort(arr):
    swap = False
    while not swap:
        swap = True
        for i in range(1,len(arr)):
            if(arr[i-1] > arr[i]):
                swap = False
                temp = arr[i-1]
                arr[i-1] = arr [i]
                arr[i] = temp
    return arr



print("Array after sorted: ")
print(bubble_sort(arr))

File: Sorting Algorithms/Bubble Sort/Diff Approaches/Java/src/Bubble_Sort_alt1.java
public class Bubble_Sort_alt1 {
    public static void main(String[] args) {

        int [] arr = {4,5,7,2,4,7,1,9,7};

        System.out.println("Array before getting sorted: ");
        printArr(arr);

        System.out.println("\nArray after getting sorted: ");
        printArr(bubbleSort(arr));

    }

    private static int[] bubbleSort(int[] arr) {
        boolean swap = false;
        while (!swap){
            swap = true;
            for(int i = 1; i <= arr.length - 1; i++){
                if(arr[i-1] > arr[i]){
                    swap = false;
                    int temp = arr[i-1];
                    arr[i-1] = arr[i];
                    arr[i] = temp;
                }
            }
        }
        return arr;
    }

    private static void printArr(int[] arr) {
        for(int i = 0; i <= arr.length - 1; i++){
            System.out.print(arr[i] + " ");
        }
    }
}
